app/src/main/java/com/github/polybooks/core/database/implementation/SaleDatabase.kt

```diff
@@ -206,31 +201,21 @@ class SaleDatabase : SaleDatabase {
     }
 
     private fun snapshotToUser(map: HashMap<String,Any>): User {
-        Log.d("8===================================","${map}" )
-        val b1 = (map[UserFields.UID.fieldName] as Long).toInt()
-        Log.d("b1++++++++++++++++++++","${b1!!.javaClass.name}" )
-        val b2 = map[UserFields.PSEUDO.fieldName] as String
-        Log.d("b2++++++++++++++++++++","${b2}" )
-        return LoggedUser(
-                b1 as Int,
-                b2
-        )
+        val uid = (map[UserFields.UID.fieldName] as Long).toInt()
+        val pseudo = map[UserFields.PSEUDO.fieldName] as String
 
+        return LoggedUser(uid, pseudo)
     }
 
     private fun snapshotToSale(snapshot: DocumentSnapshot): Sale {
-        Log.d("6===================================","${snapshot}" )
-        //println("6=========================${snapshot}")
-        val a1 = snapshotToBook(snapshot.get(SaleFields.BOOK.fieldName)!! as HashMap<String, Any>)
-        Log.d("a1++++++++++++++++++++","${a1}" )
-        val a2 = snapshotToUser(snapshot.get(SaleFields.SELLER.fieldName)!! as HashMap<String, Any>)
-        Log.d("a2++++++++++++++++++++","${a2}" )
-        val a3 = snapshot.getLong(SaleFields.PRICE.fieldName)!!.toFloat()
-        Log.d("a3++++++++++++++++++++","${a3}" )
+        val book = snapshotToBook(snapshot.get(SaleFields.BOOK.fieldName)!! as HashMap<String, Any>)
+        val seller = snapshotToUser(snapshot.get(SaleFields.SELLER.fieldName)!! as HashMap<String, Any>)
+        val price = snapshot.getLong(SaleFields.PRICE.fieldName)!!.toFloat()
+
         return Sale(
-                a1,//snapshotToBook(snapshot.get(SaleFields.BOOK.fieldName)!! as HashMap<String, Any>),
-                a2,//snapshotToUser(snapshot.get(SaleFields.SELLER.fieldName)!! as HashMap<String, Any>),
-                a3,//snapshot.getLong(SaleFields.PRICE.fieldName)!!.toFloat(),
+                book,
+                seller,
+                price,
                 BookCondition.valueOf(snapshot.getString(SaleFields.CONDITION.fieldName)!!),
                 Timestamp(snapshot.getTimestamp(SaleFields.PUBLICATION_DATE.fieldName)!!.toDate()),
                 SaleState.valueOf(snapshot.getString(SaleFields.STATE.fieldName)!!),
```

Comment:
Is there a reason why you left half the fields as "non-variable" (and half the fields pre-computed)?